File: leetcode/485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 9/6/2026, 2:56:01 PM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_cnt = 0
        cnt = 0

        for i in range(len(nums)):
            if nums[i] == 1:
                cnt += 1
                max_cnt = max(max_cnt, cnt)
            else:
                cnt = 0

        return max_cnt

File: leetcode/485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 9/6/2026, 2:56:05 PM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count, max_count;
        count = max_count = 0;

        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] == 1)
            {
                count++;
            }
            else
            {
                count = 0;
            }
            if(count > max_count)
            {
                max_count = count;
            }
        }
        return max_count;
    }
};